docs/architecture/adr-064-batch-verification.md

```diff
@@ -0,0 +1,82 @@
+# ADR 064: Batch Verification
+
+## Changelog
+
+- January 28, 2021: Created (@marbar3778)
+
+## Context
+
+Tendermint uses public private key cryptography for validator signing. When a block is proposed and voted on validators sign a message representing acceptance or rejection of a block. These signatures are also used to verify previous blocks are correct if a node is syncing. Currently, Tendermint requires each signature to be verified individually, this leads to a slow down of block times.
+
+## Alternative Approaches
+
+- Signature aggregation
+  - Signature aggregation is an alternative to batch verification. Signature aggregation leads to fast verification and smaller block sizes. At the time of writing this ADR there is on going work to enable signature aggregation in Tendermint. The reason why we have opted to not introduce it at this time is because it will slow down verification times. One of the two benefits can be done, aggregation, but verification would suffer because the software would need to verify each signature individually due to their uniqueness, then aggregate them. For example if we were to implement signature aggregation with BLS, there could be a potential slow down of 10x-100x in verification speeds.
+
+## Decision
+
+Adopt Batch Verification.
+
+## Detailed Design
+
+A new interface will be introduced. This interface will have three methods `NewBatchVerifier`, `Add` and `VerifyBatch`.
+
+```go
+type BatchVerification interface {
+  NewBatchVerifier() BatchVerification // NewBatchVerifier create a new verifier where keys, signatures and messages can be added as entries
```

Comment:
```suggestion
  New() BatchVerifier // NewBatchVerifier create a new verifier where keys, signatures and messages can be added as entries
```

Reply:
I am more inclined to leave it as `NewBatchVerifier`. This will live alongside the public & private keys, calling it new wont tell the user if they are creating a new public key, private key or batchverifier